File: Unit_tests_python/EX-03-zipf_law/Claude.py
import re
from collections import Counter

def zipfs_law(text):
    # Remove punctuation and make lowercase
    text = re.sub(r'[^\w\s]', '', text.lower())
    
    # Split into words
    words = text.split()
    
    # Count word frequencies
    word_counts = Counter(words)
    
    # Get top 10 words
    top_10 = word_counts.most_common(10)
    
    # Extract the words only
    top_10_words = [word for word, count in top_10]
    
    return top_10_words

File: Unit_tests_lua/EX-03-zipf_law/Claude.lua
function zipfs_law(str)
    -- Remove punctuation and make lowercase
    str = string.gsub(string.lower(str), "%p", "")
    
    -- Split into words
    local words = {}
    for word in string.gmatch(str, "%w+") do
      table.insert(words, word) 
    end
    
    -- Count frequency of each word
    local freq = {}
    for _, word in ipairs(words) do
      freq[word] = (freq[word] or 0) + 1
    end
  
    -- Sort by frequency
    local sorted_freq = {}
    for word, count in pairs(freq) do
      table.insert(sorted_freq, {word, count})
    end
    table.sort(sorted_freq, function(a, b) return a[2] > b[2] end)
  
    -- Return top 10
    local top10 = {}
    for i = 1, math.min(10, #sorted_freq) do
      table.insert(top10, sorted_freq[i][1])
    end
    return top10
  end